metagraph/src/cli/align.cpp

```diff
@@ -393,12 +394,12 @@ int align_to_graph(Config *config) {
     ThreadPool thread_pool(get_num_threads());
     std::mutex print_mutex;
 
-    if (graph->get_mode() == DeBruijnGraph::PRIMARY) {
-        logger->trace("Primary graph wrapped into canonical");
-        graph = std::make_shared<CanonicalDBG>(graph);
-    }
-
     if (config->map_sequences) {
+        if (graph->get_mode() == DeBruijnGraph::PRIMARY) {
+            logger->trace("Primary graph wrapped into canonical");
+            graph = std::make_shared<CanonicalDBG>(graph);
+        }
```

Comment:
Why only for `map_sequences` and not always wrap the graph?

Reply:
I'll put a comment here to explain this. Later in the alignment code, I create a separate `CanonicalDBG` wrapper for each thread to avoid contention when accessing cached data.

Reply:
What about re-wrapping it there?

Reply:
The graph is wrapped in all cases. It's just that for the alignment (non-`map_sequences`) case, I defer the wrapping until after a thread is spawned so I can keep a separate `CanoncialDBG` for each thread.

If the graph is wrapped from the beginning, then I'd have to do a `dynamic_cast` check, unwrap, then rewrap in the alignment section, which is a bit ugly.

Reply:
I see. But that might be not the only place where you want to isolate caches of two graph views.
A more general approach for handling such cases is wrapping the graph first and then create a new copy of the wrapped graph for each thread where you create it:
```
graph
if (graph is wrapped) {
    graph = new canonical_dbg(graph.base_graph());
}
use graph
```

Reply:
I didn't do this originally because `CanonicalDBG` doesn't have a copy constructor. I added one.